invenio_github/receivers.py

```diff
@@ -49,39 +57,102 @@ def run(self, event):
             the rest of the processing to a Celery task which will be mainly
             accessing the GitHub API.
         """
-        repo_id = event.payload['repository']['id']
-
-        # Ping event - update the ping timestamp of the repository
-        if 'hook_id' in event.payload and 'zen' in event.payload:
-            repository = Repository.query.filter_by(
-                github_id=repo_id
-            ).one()
-            repository.ping = datetime.utcnow()
-            db.session.commit()
-            return
-
-        # Release event
-        action = event.payload.get('action')
-        is_draft_release = event.payload.get('release', {}).get('draft')
-        is_release_event = (
-            action in ('published', 'released', 'created')
-            and not is_draft_release
+        try:
+            self._handle_event(event)
+        except Exception as e:
+            # Event failed to be processed and error was not handled yet
+            if not event.response or event.response_code < 400:
+                event.response = {"status": 500, "message": str(e)}
+                event.response_code = 500
+            # Other cases were already handled (e.g. response/response_code were set by the event handler)
+
+    def _handle_event(self, event):
+        """Handles an incoming github event."""
+        action = event.payload.get("action")
+        is_draft_release = event.payload.get("release", {}).get("draft")
+
+        # Draft releases do not create releases on invenio
+        is_create_release_event = (
+            action in ("published", "released", "created") and not is_draft_release
         )
-        if is_release_event:
+
+        if is_create_release_event:
+            self._handle_create_release(event)
+        else:
+            # TODO other events (e.g. ping, draft release) are discarded
+            pass
+
+    def _handle_create_release(self, event):
+        """Creates a release in invenio."""
+
+        def _create_release(event):
```

Comment:
This seems to have the full logic of the outer function, why do we need a function inside a function.